Suggested fix for #1047: CachedReferenceField DBRefs bug

amitlicht opened the pull request:
CachedReferenceField creates DBRef on to_python, but can't save them on to_mongo.

Dereferencing DBRef to document type before returning it from to_python.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/mongoengine/mongoengine/1048)
<!-- Reviewable:end -->

MRigal commented:
Hi @amitlicht 
Thanks for the PR! Don't pay attention to the fail tests, it is Travis internal. Could you please convert your #1047 issue into a proper test case that you could push to your branch? Having the fix without a testcase is not enough. Also consider adding a line to the Changelog and adding yourself to the authors :-)

amitlicht commented:
Hi @MRigal 
I added a test case into tests/fields/fields.py. It was tested before and after my change and indeed demonstrates the bug & its fix. I also added a changelog line & added myself to the authors as you suggested :). Anything else?

MRigal commented:
Thanks @amitlicht
